crates/oxc_linter/src/rules/typescript/prefer_ts_expect_error.rs

```diff
@@ -0,0 +1,229 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use oxc_ast::Comment;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::fixer::Fix;
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "typescript-eslint(prefer-ts-expect-error): Enforce using `@ts-expect-error` over `@ts-ignore`"
+)]
+#[diagnostic(
+    severity(warning),
+    help("Use \"@ts-expect-error\" to ensure an error is actually being suppressed.")
+)]
+struct PreferTsExpectErrorDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferTsExpectError;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforce using @ts-expect-error over @ts-ignore.
+    ///
+    /// ### Why is this bad?
+    /// TypeScript allows you to suppress all errors on a line by placing a comment starting with @ts-ignore or @ts-expect-error immediately before the erroring line.
+    /// The two directives work the same, except @ts-expect-error causes a type error if placed before a line that's not erroring in the first place.
+    ///
+    /// This means it's easy for @ts-ignores to be forgotten about, and remain in code even after the error they were suppressing is fixed.
+    /// This is dangerous, as if a new error arises on that line it'll be suppressed by the forgotten about @ts-ignore, and so be missed.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // @ts-ignore
+    /// const str: string = 1;
+    ///
+    /// /**
+    /// * Explaining comment
+    /// *
+    /// * @ts-ignore */
+    /// const multiLine: number = 'value';
+    /// ```
+    PreferTsExpectError,
+    correctness
```

Comment:
Should this be `pedantic`?

Reply:
I don't really understand what this field is for - is there documentation somewhere for it?